tests/test_shard.py

```diff
@@ -37,14 +38,14 @@ def sync_model_params(model):
 
 def gather_grad(model, param_path_and_gather_axis):
     world_size = dist.get_world_size()
-    rank = dist.get_rank()
+    local_rank = int(os.environ["LOCAL_RANK"])
```

Comment:
Out of curiosity, what's the different between these two?

Reply:
local_rank means the rank on the local node. For example, rank = 8 -> local_rank = 0 when each node has 8 gpus.

Reply:
Oh nvm. I missed that `get_rank()` returns the global rank.

Reply:
Does this environment variable always exist?